source/agora/node/Registry.d

```diff
@@ -413,7 +413,9 @@ public class NameRegistry: NameRegistryAPI
                         goto BAILOUT;
                     }
                     // Ignore the DO bit for now
-                    payloadSize = max(opt.payloadSize(), ushort(512));
+                    // `min` is to prevent DOS attack (request with huge payload size)
+                    payloadSize = min(max(opt.payloadSize(), ushort(512)),
+                        OPTRR.init.payloadSize());
```

Comment:
Actually, looking at the code again, `responseOPT.payloadSize()` is more correct (same value in the end).
`OPT` is a way to advertise the max payload size for both clients and server. A server that would send a record with a payload higher than its own OPT `payloadSize()` would be buggy.

Reply:
`payloadSize` is not responded back, always `responseOPT` is used and its `OPTRR.init`

Reply:
Yes but if we ever change the OPT we respond, using `responseOPT.payloadSize()` will still be valid, while `OPTRR.init` will not.

Reply:
Updated